packages/react-next/src/components/ChoiceGroup/ChoiceGroup.base.tsx

```diff
@@ -1,128 +1,141 @@
+    const changeCallbacks: { [key: string]: IChoiceGroupOptionProps['onBlur'] } = {};
+
+    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties, [
+      'onChange',
+      'className',
+      'required',
+    ]);
 
-  constructor(props: IChoiceGroupProps) {
-    super(props);
+    const classNames = getClassNames(styles!, {
+      theme: theme!,
+      className,
+      optionsContainIconOrImage: options.some(option => !!(option.iconProps || option.imageSrc)),
+    });
 
-    initializeComponentRef(this);
+    const ariaLabelledBy = props.ariaLabelledBy || (label ? labelId : props['aria-labelledby']);
 
-    if (process.env.NODE_ENV !== 'production') {
-      warnDeprecations('ChoiceGroup', props, { onChanged: 'onChange' });
-      warnMutuallyExclusive('ChoiceGroup', props, {
-        selectedKey: 'defaultSelectedKey',
-      });
-    }
+    /**
+     * Returns `selectedKey` if provided, or the key of the first option with the `checked` prop set.
+     */
+    const getKeyChecked = (): string | number | undefined => {
+      if (props.selectedKey !== undefined) {
+        return props.selectedKey;
+      }
+      // eslint-disable-next-line deprecation/deprecation
+      const optionsChecked = options.filter((option: IChoiceGroupOption) => option.checked);
+      return optionsChecked[0] && optionsChecked[0].key;
+    };
 
-    const { defaultSelectedKey, options = [] } = props;
     const validDefaultSelectedKey =
-      !_isControlled(props) &&
+      !isControlled(props, 'selectedKey') &&
       defaultSelectedKey !== undefined &&
       options.some(option => option.key === defaultSelectedKey);
 
-    this.state = {
-      keyChecked: validDefaultSelectedKey ? defaultSelectedKey : this._getKeyChecked(props),
-    };
+    const [keyChecked, setKeyChecked] = React.useState(validDefaultSelectedKey ? defaultSelectedKey : getKeyChecked());
+
+    const [keyFocused, setKeyFocused] = React.useState<string | number>();
 
-    this._id = getId('ChoiceGroup');
-    this._labelId = getId('ChoiceGroupLabel');
-  }
-
-  /**
-   * Gets the current checked option.
-   */
-  public get checkedOption(): IChoiceGroupOption | undefined {
-    const { options = [] } = this.props;
-    return find(options, (value: IChoiceGroupOption) => value.key === this.state.keyChecked);
-  }
-
-  public componentDidUpdate(prevProps: IChoiceGroupProps, prevState: IChoiceGroupState): void {
-    // Only update if a new props object has been passed in (don't care about state updates)
-    if (prevProps !== this.props) {
-      const newKeyChecked = this._getKeyChecked(this.props);
-      const oldKeyChecked = this._getKeyChecked(prevProps);
-
-      if (newKeyChecked !== oldKeyChecked) {
-        this.setState({
-          keyChecked: newKeyChecked,
-        });
+    const onFocus = (key: string) => {
+      // This extra mess is necessary because React won't pass the `key` prop through to ChoiceGroupOption
```

Comment:
Ah this is indeed a mess. i think we can and should just fix this.

instead of having `key: string` in `IChoiceGroupOptionProps` , we should change it to `optionKey: string`.
that way we can read/set the key not using the preserved prop `key`. 

this way, we can clean up the code here to simply be:

```
const onFocus = (ev: React.FocusEvent<HTMLElement | HTMLInputElement>, option: IChoiceGroupOption) => setKeyFocused(option.optionKey)
```
similarly for `onChange` / `changeCallbacks`